service/history/replication/dlq_handler.go

```diff
@@ -189,7 +202,7 @@ func (r *dlqHandlerImpl) MergeMessages(
 	)
 
 	for _, task := range tasks {
-		if _, err := r.taskExecutor.execute(
+		if _, err := r.taskExecutors[sourceCluster].execute(
 			sourceCluster,
```

Comment:
nit: seems like we can move the `sourceCluster` to the constructor of replication task executor.

Reply:
moved source cluster to constructor